### PR TITLE
Prevent rename to existing name

### LINKED ISSUE #912: renaming files/folder into a name of an existing file/folder overwrites the destination
### Steps to reproduce
1. have a file `file1.txt` with content `file1` and a file `file2.txt` with content `file2`
2. rename `file2.txt` into `file1.txt`
3. with a different client check the content of `file1.txt`

### Expected behaviour
rename not possible, error shown on the UI

### Actual behaviour
rename happens, the file with the destination name is overwritten

#### Browser log
image

### PR DESCRIPTION
**Pull request by LukasHirt**
## Description
Added condition into rename function to check if file name already exists.

## Related Issue

- Fixes #912 

## How Has This Been Tested?

- test environment: Manually

## Screenshots (if appropriate):
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: apps/files/src/mixins.js [hunk 1]
```diff
@@ -56,20 +56,31 @@ export default {
...
-        this.renameFile({
-          client: this.$client,
-          file: this.selectedFile,
-          newValue: item
+        return
+      }
+
+      if (item.includes('/')) {
+        this.showNotification({
+          title: this.$gettext(`Renaming of ${this.selectedFile.name} failed. The file name cannot contain a "/"`),
```
Comment (DeepDiver1975) [inline apps/files/src/mixins.js]: please see #1088 on how to properly interpolate translated strings

Comment (LukasHirt) [reply apps/files/src/mixins.js]: The interpolation then has the problem with special characters - #911

Comment (DeepDiver1975) [reply apps/files/src/mixins.js]: I see ....

Comment (DeepDiver1975) [reply apps/files/src/mixins.js]: $gettextInterpolate has a third argument to disable escaping - we need to see if this opens any sec issues ....

Comment (DeepDiver1975) [reply apps/files/src/mixins.js]: see https://github.com/owncloud/phoenix/pull/1088/files#diff-c48186b3f0ec2249fa2211b9a5ae18beR64

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFavorites</strong> failed. Please find the screenshots inside ...</summary><p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFolders</strong> failed. Please find the screenshots inside ...</summary><p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIUpload</strong> failed. Please find the screenshots inside ...</summary><p>














</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary><p>


















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary><p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary><p>




























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary><p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary><p>






</p></details>

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2849/

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2849/

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2849/

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2849/

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2849/

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary><p>






</p></details>

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2849/

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary><p>






</p></details>

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2855/

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2855/

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2855/

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2855/

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary><p>






</p></details>

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2855/

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2855/

**Comment by LukasHirt** [comment]
@DeepDiver1975 Tests has finally passed - review pls 🙏

**Pull request merged**